transport/http/checksum_middleware_test.go

```diff
@@ -0,0 +1,57 @@
+package http
```

Comment:
Would be great to have a test that ensures the `Stream` is `Rewindable`, not empty, and has original length after after the checksum is computed.